### PR TITLE
ci: simplify contracts bedrock tests job

### PR DESCRIPTION
**Pull request by smartcontracts**
Simplifies the contracts-bedrock-tests job so that it takes more parameters but makes the test command unified. Also replaces the complex script for checking for modified fuzz tests and simply heavily fuzzes all tests within any test files that have changed.

### CODE REVIEW

**Review by ajsutton — APPROVED** [review]

**Review by maurelian — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -609,30 +622,36 @@ jobs:
...
       - run:
-          name: run tests
-          command: <<parameters.test_command>>
+          name: Run tests
+          command: |
+            TEST_FILES=$(<<parameters.test_list>>)
+            TEST_FILES=$(echo "$TEST_FILES" | circleci tests split --split-by=timings)
+            TEST_FILES=$(echo "$TEST_FILES" | sed 's|^test/||')
+            MATCH_PATH="./test/{$(echo "$TEST_FILES" | paste -sd "," -)}"
+            export FOUNDRY_INVARIANT_RUNS=<<parameters.test_fuzz_runs>>
```
Comment (mds1) [inline .circleci/config.yml:647]: Invariant tests are much slower than regular fuzz tests, I don't think we want to use the same amount of runs for both. For invariant tests it's better to have fewer test runs with a very large depth (longer call sequence). See https://github.com/foundry-rs/foundry/issues/7848. Not a blocker though, we can fix this and pull out the invariant config in a separate PR
